spring-ai-template-st/src/main/java/org/springframework/ai/template/st/StTemplateRenderer.java

```diff
@@ -130,7 +131,7 @@ private ST createST(String template) {
 	 * @param templateVariables the provided variables
 	 * @return set of missing variable names, or empty set if none are missing
 	 */
-	private Set<String> validate(ST st, Map<String, Object> templateVariables) {
+	private Set<String> validate(ST st, Map<String, @Nullable Object> templateVariables) {
```

Comment:
Should probably be `Map<String, ? extends @Nullable Object>` to allow both `Map<String, @Nullable Object>` and `Map<String, Object>` (flexible nullability).